examples/multiple_viewer_widget.py

```diff
@@ -0,0 +1,427 @@
+
+def copy_layer(layer: Layer, name: str = ""):
+    res_layer = deepcopy(layer)
+    # this deepcopy is not optimal for layers and images layers
+    if isinstance(layer, (Image, Labels)):
+        res_layer.data = layer.data
+
+    res_layer.metadata["viewer_name"] = name
+
+    res_layer.events.disconnect()
+    res_layer.events.source = res_layer
+    for emitter in res_layer.events.emitters.values():
+        emitter.disconnect()
+        emitter.source = res_layer
+    return res_layer
+
+
+def get_property_names(layer: Layer):
+    klass = layer.__class__
+    res = []
+    for event_name in layer.events.emitters:
+        if event_name in ("thumbnail", "name"):
+            continue
+        if (
+            isinstance(getattr(klass, event_name, None), property)
+            and getattr(klass, event_name).fset is not None
+        ):
+            res.append(event_name)
+    return res
+
+
+def center_cross_on_mouse(viewer: napari.Viewer):
+    """move the cross to the mouse position"""
+
+    print(viewer.title, viewer.cursor.position)
+    # FIXME times to time wrong viewer is selected
+
+    if not getattr(viewer, "_mouse_over_canvas", True):
+        # There is no way fo napari 0.4.15 to check if mose is over sending canvas. 
+        show_info("Mouse is not over the canvas. You may need to click on the canvas.")
+        return
+
+    viewer.dims.current_step = tuple(
+        np.round(
+            [
+                max(min_, min(p, max_)) / step
+                for p, (min_, max_, step) in zip(
+                    viewer.cursor.position, viewer.dims.range
+                )
+            ]
+        ).astype(int)
+    )
+
+
+action_manager.register_action(
+    name='napari:move_point',
+    command=center_cross_on_mouse,
+    description='Move dims point to mouse position',
+    keymapprovider=ViewerModel,
+)
+
+action_manager.bind_shortcut('napari:move_point', 'C')
+
+
+class own_partial:
+    """
+    Workaround for deepcopy do not copy partial functions
+    (Qt widgets are not serializable)
+    """
+
+    def __init__(self, func, *args, **kwargs):
+        self.func = func
+        self.args = args
+        self.kwargs = kwargs
+
+    def __call__(self, *args, **kwargs):
+        return self.func(*(self.args + args), **{**self.kwargs, **kwargs})
+
+    def __deepcopy__(self, memodict={}):
+        return own_partial(
+            self.func,
+            *deepcopy(self.args, memodict),
+            **deepcopy(self.kwargs, memodict),
+        )
+
+
+class QtViewerWrap(QtViewer):
+    def __init__(self, main_viewer, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.main_viewer = main_viewer
+
+    def _qt_open(
+        self,
+        filenames: list,
+        stack: bool,
+        plugin: str = None,
+        layer_type: str = None,
+        **kwargs,
+    ):
+        """for drag and drop open files"""
+        self.main_viewer.window._qt_viewer._qt_open(filenames, stack, plugin, layer_type, **kwargs)
```

Comment:
this is a workaround to drag and drop on an additional canvas. But I'm not proud of this. Maybe I should just block dropping files on additional viewers?

Reply:
Two things. First is what you write. The second one is that there is a need to subclass `QtViewer` to solve this problem.